tests/test_modifiers.py

```diff
@@ -14,138 +11,30 @@
-    assert pyhf.modifiers.registry['myConstrainedModifier'].is_constrained is True
-    del pyhf.modifiers.registry['myConstrainedModifier']
-
-
-# we make sure decorate can use auto-naming
-def test_modifier_name_auto():
-    from pyhf.modifiers import modifier
-
-    @modifier
-    class myCustomModifier:
-        @classmethod
-        def required_parset(cls, sample_data, modifier_data):
-            pass
-
-    assert inspect.isclass(myCustomModifier)
-    assert 'myCustomModifier' in pyhf.modifiers.registry
-    assert pyhf.modifiers.registry['myCustomModifier'] == myCustomModifier
-    del pyhf.modifiers.registry['myCustomModifier']
-
-
-# we make sure decorate can use auto-naming with keyword arguments
-def test_modifier_name_auto_withkwargs():
-    from pyhf.modifiers import modifier
-
-    @modifier(name=None, constrained=False)
-    class myCustomModifier:
-        @classmethod
-        def required_parset(cls, sample_data, modifier_data):
-            pass
-
-    assert inspect.isclass(myCustomModifier)
-    assert 'myCustomModifier' in pyhf.modifiers.registry
-    assert pyhf.modifiers.registry['myCustomModifier'] == myCustomModifier
-    del pyhf.modifiers.registry['myCustomModifier']
-
-
-# we make sure decorate allows for custom naming
-def test_modifier_name_custom():
-    from pyhf.modifiers import modifier
-
-    @modifier(name='myCustomName')
-    class myCustomModifier:
-        @classmethod
-        def required_parset(cls, sample_data, modifier_data):
-            pass
-
-    assert inspect.isclass(myCustomModifier)
-    assert 'myCustomModifier' not in pyhf.modifiers.registry
-    assert 'myCustomName' in pyhf.modifiers.registry
-    assert pyhf.modifiers.registry['myCustomName'] == myCustomModifier
-    del pyhf.modifiers.registry['myCustomName']
-
-
-# we make sure decorate raises errors if passed more than one argument, or not a string
-def test_decorate_with_wrong_values():
-    from pyhf.modifiers import modifier
-
-    with pytest.raises(ValueError):
-
-        @modifier('too', 'many', 'args')
-        class myCustomModifier:
-            pass
-
-    with pytest.raises(TypeError):
-
-        @modifier(name=1.5)
-        class myCustomModifierTypeError:
-            pass
-
-    with pytest.raises(ValueError):
-
-        @modifier(unused='arg')
-        class myCustomModifierValueError:
-            pass
-
-
-# we catch name clashes when adding duplicate names for modifiers
-def test_registry_name_clash():
-    from pyhf.modifiers import modifier
-
-    with pytest.raises(KeyError):
-
-        @modifier(name='histosys')
-        class myCustomModifierKeyError:
-            pass
-
-    with pytest.raises(KeyError):
-
-        class myCustomModifier:
-            @classmethod
-            def required_parset(cls, sample_data, modifier_data):
-                pass
-
-        pyhf.modifiers.add_to_registry(myCustomModifier, 'histosys')
+def test_shapefactor_build():
```

Comment:
coverage is lost here as we're only testing that we can build shapefactor and none of the others? This would be great if we could parametrize it for tests.